pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -1154,3 +1226,563 @@ func (testVals *testGKENetworkParamSetController) doesGNPFinalizerExist(ctx cont
+				IsValid: true,
+			},
+		},
+		{
+			name: "Default network with new PodIPv4Ranges",
+			network: &networkv1.Network{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkv1.DefaultPodNetworkName,
+				},
+				Spec: networkv1.NetworkSpec{
+					Type: networkv1.L3NetworkType,
+				},
+			},
+			params: &networkv1.GKENetworkParamSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkv1.DefaultPodNetworkName,
+				},
+				Spec: networkv1.GKENetworkParamSetSpec{
+					VPC:           defaultTestNetworkName,
+					PodIPv4Ranges: &networkv1.SecondaryRanges{RangeNames: []string{defaultPodRange, newPodRange}},
+				},
+				Status: networkv1.GKENetworkParamSetStatus{
+					PodCIDRs: &networkv1.NetworkRanges{
+						CIDRBlocks: []string{defaultPodCIDR, newPodCIDR},
+					},
+				},
+			},
+			originalParams: &networkv1.GKENetworkParamSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: networkv1.DefaultPodNetworkName,
+				},
+				Spec: networkv1.GKENetworkParamSetSpec{
+					VPC:           defaultTestNetworkName,
+					PodIPv4Ranges: &networkv1.SecondaryRanges{RangeNames: []string{defaultPodRange}},
+				},
+				Status: networkv1.GKENetworkParamSetStatus{
+					PodCIDRs: &networkv1.NetworkRanges{
+						CIDRBlocks: []string{defaultPodCIDR},
+					},
+				},
+			},
+			want: &gnpNetworkCrossValidation{
+				IsValid:      false,
+				ErrorReason:  networkv1.GNPParamsNotReady,
+				ErrorMessage: "New Pod ranges in default VPC requires network CIDRs update",
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx, stop := context.WithCancel(context.Background())
+			defer stop()
+			testVals := setupGKENetworkParamSetController(ctx)
+			testVals.runGKENetworkParamSetController(ctx)
+
+			network, err := testVals.networkClient.NetworkingV1().Networks().Create(ctx, tc.network, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create Network: %v", err)
+			}
+
+			originalParams, err := testVals.networkClient.NetworkingV1().GKENetworkParamSets().Create(ctx, tc.originalParams, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create GKENetworkParamSet: %v", err)
+			}
+
+			params, err := testVals.networkClient.NetworkingV1().GKENetworkParamSets().Update(ctx, tc.params, metav1.UpdateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create GKENetworkParamSet: %v", err)
+			}
+
+			got := crossValidateNetworkAndGnp(network, params, originalParams)
+			if !reflect.DeepEqual(got, tc.want) {
+				t.Fatalf("crossValidateNetworkAndGnp(%+v) returns %v but want %v", tc.name, got, tc.want)
+			}
+		})
+	}
+}
+
+func TestSameStringSlice(t *testing.T) {
+	tests := []struct {
+		name string
+		x    []string
+		y    []string
+		want bool
+	}{
+		{
+			name: "not same len",
+			x:    []string{"a"},
+			y:    []string{""},
+			want: false,
+		},
+		{
+			name: "same order",
+			x:    []string{"ab", "bc"},
+			y:    []string{"ab", "bc"},
+			want: true,
+		},
```

Comment:
add a test case with  x= ["a", "a","b"] and y = ["a","b","a"] want true

Reply:
Done.